.github/workflows/label-prs.yml

```diff
@@ -0,0 +1,26 @@
+name: 'Pull Request Labeler'
+on:
+  - pull_request_target
```

Comment:
By default this runs for `opened`, `synchronize`, or `reopened` (see [the docs](https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#pull_request_target)). I'd be inclined to limit this to just `opened` so it runs just a single time when the PR is opened. If a human decides to remove a label, I don't want this bot adding it again anytime the PR is updated.

Reply:
> f the bot gets a label wrong, would the correct response be to update the rules?

For code-region changes, I think this is. If it mistags a region, the rules are wrong.

Reply:
IMO that conflicts with this:

> I feel like there are diminishing returns for some of these one-of files. I'd rather not do this, and focus on having the rules hit 90% of PRs.

If we're not going to leave the final call to humans, we _must_ make the rules as accurate as possible for everything, including "one-off files".

Reply:
The labeler will only add labels on sync, never remove them. https://github.com/actions/labeler/issues/112

So if we miss a file, a human can add it.

If it tags a file it shouldn't, then it would keep trying to readd the tag.

Reply:
That's inconsistent with the docs for `sync-labels` on https://github.com/actions/labeler:

> Whether or not to remove labels when matching files are reverted or no longer changed by the PR

Is that just out of date?

Reply:
Yeah, in the README under "v5 breaking changes" it links to that issue. https://github.com/actions/labeler?tab=readme-ov-file#breaking-changes-in-v5

Reply:
I think you're misunderstanding what that issue is about. Look at [the PR](https://github.com/actions/labeler/pull/480) that fixed it. The issue is that if `sync-labels` was not specified, it would default to `true`. As far as I can tell, the behavior for `sync-labels: true` did not change in v5. The implementation clearly deletes labels when `sync-labels` is `true`: https://github.com/actions/labeler/blob/6463cdb00ee92c05bec55dffc4e1fce250301945/src/labeler.ts#L49-L50

Reply:
My apologies, I did misread that PR. I set sync-labels to False in b650d16b59f967ed737b7fea88c755cc9b0354c5.